src/bitdrift_public/protobuf/client/v1/api.proto

```diff
@@ -256,6 +256,10 @@ message UploadArtifactIntentRequest {
   // The metadata associated with the artifact. This is a binary blob that is interpreted by the server
   // based on the type_id.
   bytes metadata = 3;
+
+  // A client-generated ID that uniquely identifies the artifact. This is used to correlate the artifact
+  // with logs that reference it.
+  string artifact_id = 4 [(validate.rules).string = {min_len: 1}];
```

Comment:
I feel like the intent_uuid and the upload_uuid below can probably be used for this, but it doesn't really matter if you want to keep them independent for future proofing.

Reply:
Yeah I had the same thought but seemed nice to have it split in case these two end up diverging